src/serializer/ResidualHeapSerializer.js

```diff
@@ -718,29 +720,51 @@ export class ResidualHeapSerializer {
     return Array.from(result);
   }
 
-  // Determine if a value is effectively referenced by a single additional function.
+  isNestedOptimizedFunction(optimizedFunction: FunctionValue, parentOptimizedFunction: FunctionValue): boolean {
```

Comment:
This doesn't seem to be tied to optimized semantics. Can this be renamed to `isDefinedInsideFunction(childFunction, maybeParentFunction)` or similar?

Reply:
It is tied to nested optimized functions, this only occurs if a function is optimized inside another optimized function.

Reply:
I mean the function implementation itself is generic and would work with any two functions. It's confusing to me that "Is nested optimized function" takes an argument called "optimized function". It's like if `isNumber` argument was called `number`. This makes it hard to say what you can pass in.

Similarly, I assume that if "Is nested optimized function" returns true, it means the passed function (or, rather, one of them: the function signature doesn't tell me which one is being tested), is indeed optimized. That's what the function name claims. But in practice you can easily get a `true` even if you passed a function that wasn't optimized.

This is why I think a name like `isDefinedInsideFunction(childFunction, maybeParentFunction)` or `isDefinedInsideFunction(childFunction, maybeParentFunction)` would describe what this function does more accurately. But that's not super important, just a nitpick.

Reply:
Thanks for the detailed explanation that makes sense. I’m happy for you to rename it, I’m at the dentist right now :)